webpack.config.js

```diff
@@ -138,14 +141,17 @@ module.exports = {
 		filename: ( data ) => {
 			const { chunk } = data;
 			const { entryModule } = chunk;
-			const { rawRequest } = entryModule;
+			const { rawRequest, rootModule } = entryModule;
+
+			// When processing ESM files, the requested path
+			// is defined in `entryModule.rootModule.rawRequest`, instead of
+			// being present in `entryModule.rawRequest`.
+			// In the context of frontend view files, they would be processed
+			// as ESM if they use `import` or `export` within it.
+			const request = rootModule?.rawRequest || rawRequest;
 
-			/*
-			 * If the file being built is a Core Block's frontend file,
-			 * we build it in the block's directory.
-			 */
-			if ( rawRequest && rawRequest.includes( '/frontend.js' ) ) {
-				return `./build/block-library/blocks/[name]/frontend.js`;
+			if ( request.includes( '/view.js' ) ) {
+				return `./build/block-library/[name]/view.min.js`;
```

Comment:
I think we don't have the changes from the `trunk` that uses `.min.js` extension:

```suggestion
				return `./build/block-library/[name]/view.js`;
```

Related PR: https://github.com/WordPress/gutenberg/pull/31732.